### ISSUE TITLE
DirectoryNotEmptyException on Windows

### ISSUE THREAD

**Issue by madegroot** [GoogleContainerTools/jib#1438]
**Description of the issue**:
Unable to use the jib Gradle plug-in on Windows due to DirectoryNotEmptyException.

**Expected behavior**:
Usage of jib and jibDockerBuild on Windows

**Steps to reproduce**: 
```
D:\repos\personal-repos\gradle-init-project>gradle init --type java-library --project-name jvm-library
Select build script DSL:
  1: groovy
  2: kotlin
Enter selection (default: groovy) [1..2] 1
Select test framework:
  1: junit
  2: testng
  3: spock
Enter selection (default: junit) [1..3] 1
Source package (default: jvm.library):
```
```D:\repos\personal-repos\gradle-init-project>notepad build.gradle```
- add to the plug-in section: `id 'com.google.cloud.tools.jib' version '1.0.0'` 
- add below the plug-in section `jib.to.image = 'madegroot/helloworld'`

`D:\repos\personal-repos\gradle-init-project>notepad src\main\java\jvm\library\Library.java`
- add to class `public static void main(String[] args) { System.out.println("Hello World"); }`
```
D:\repos\personal-repos\gradle-init-project>gradlew build
D:\repos\personal-repos\gradle-init-project>gradlew jib -Djib.to.auth.username=madegroot -Djib.to.auth.password=QQQ --stacktrace

Containerizing application to madegroot/helloworld...
Base image 'gcr.io/distroless/java' does not use a specific image digest - build may not be reproducible
Executing tasks:
[====================          ] 66,7% complete
> setting up to push layers

> Task :jib FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':jib'.
> com.google.cloud.tools.jib.plugins.common.BuildStepsExecutionException: Build image failed

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':jib'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:95)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:119)
        at org.gradle.api.internal.tasks.execution.ResolvePreviousStateExecuter.execute(ResolvePreviousStateExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:93)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:45)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:94)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:67)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:49)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:315)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:305)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:175)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:101)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:49)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:43)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:355)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:134)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:129)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:202)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:193)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:129)
        at org.gradle.execution.plan.DefaultPlanExecutor.process(DefaultPlanExecutor.java:74)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph.executeWithServices(DefaultTaskExecutionGraph.java:178)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph.execute(DefaultTaskExecutionGraph.java:154)
        at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:41)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:40)
        at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:24)
        at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:46)
        at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:49)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:40)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:33)
        at org.gradle.initialization.DefaultGradleLauncher$ExecuteTasks.run(DefaultGradleLauncher.java:383)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:301)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:293)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:175)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.initialization.DefaultGradleLauncher.runTasks(DefaultGradleLauncher.java:247)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:159)
        at org.gradle.initialization.DefaultGradleLauncher.executeTasks(DefaultGradleLauncher.java:134)
        at org.gradle.internal.invocation.GradleBuildController$1.execute(GradleBuildController.java:58)
        at org.gradle.internal.invocation.GradleBuildController$1.execute(GradleBuildController.java:55)
        at org.gradle.internal.invocation.GradleBuildController$3.create(GradleBuildController.java:82)
        at org.gradle.internal.invocation.GradleBuildController$3.create(GradleBuildController.java:75)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:183)
        at org.gradle.internal.work.StopShieldingWorkerLeaseService.withLocks(StopShieldingWorkerLeaseService.java:40)
        at org.gradle.internal.invocation.GradleBuildController.doBuild(GradleBuildController.java:75)
        at org.gradle.internal.invocation.GradleBuildController.run(GradleBuildController.java:55)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:31)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:58)
        at org.gradle.tooling.internal.provider.ValidatingBuildActionRunner.run(ValidatingBuildActionRunner.java:32)
        at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:39)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:49)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:44)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:315)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:305)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:175)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:101)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:44)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:49)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:46)
        at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:78)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:46)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:31)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:42)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:28)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:78)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:52)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:59)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:36)
        at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:68)
        at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:38)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:37)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:26)
        at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:43)
        at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:29)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:60)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:32)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:55)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:41)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:48)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:32)
        at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:55)
        at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:207)
        at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:403)
        at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:376)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:37)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:23)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:369)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:299)
        at org.gradle.launcher.Main.doAction(Main.java:36)
        at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:60)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:37)
        at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
        at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:31)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:108)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:62)
Caused by: org.gradle.internal.UncheckedException: com.google.cloud.tools.jib.plugins.common.BuildStepsExecutionException: Build image failed
        at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:67)
        at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:41)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:76)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:48)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:41)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:704)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:671)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$2.run(ExecuteActionsTaskExecuter.java:284)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:301)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:293)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:175)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:273)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:258)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$200(ExecuteActionsTaskExecuter.java:67)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:145)
        at org.gradle.internal.execution.impl.steps.ExecuteStep.execute(ExecuteStep.java:49)
        at org.gradle.internal.execution.impl.steps.CancelExecutionStep.execute(CancelExecutionStep.java:34)
        at org.gradle.internal.execution.impl.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:69)
        at org.gradle.internal.execution.impl.steps.TimeoutStep.execute(TimeoutStep.java:49)
        at org.gradle.internal.execution.impl.steps.CatchExceptionStep.execute(CatchExceptionStep.java:33)
        at org.gradle.internal.execution.impl.steps.CreateOutputsStep.execute(CreateOutputsStep.java:50)
        at org.gradle.internal.execution.impl.steps.SnapshotOutputStep.execute(SnapshotOutputStep.java:43)
        at org.gradle.internal.execution.impl.steps.SnapshotOutputStep.execute(SnapshotOutputStep.java:29)
        at org.gradle.internal.execution.impl.steps.CacheStep.executeWithoutCache(CacheStep.java:134)
        at org.gradle.internal.execution.impl.steps.CacheStep.lambda$execute$3(CacheStep.java:83)
        at org.gradle.internal.execution.impl.steps.CacheStep.execute(CacheStep.java:82)
        at org.gradle.internal.execution.impl.steps.CacheStep.execute(CacheStep.java:36)
        at org.gradle.internal.execution.impl.steps.PrepareCachingStep.execute(PrepareCachingStep.java:33)
        at org.gradle.internal.execution.impl.steps.StoreSnapshotsStep.execute(StoreSnapshotsStep.java:38)
        at org.gradle.internal.execution.impl.steps.StoreSnapshotsStep.execute(StoreSnapshotsStep.java:23)
        at org.gradle.internal.execution.impl.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:96)
        at org.gradle.internal.execution.impl.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:89)
        at org.gradle.internal.execution.impl.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:52)
        at org.gradle.internal.execution.impl.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:36)
        at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:34)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:91)
        ... 108 more
Caused by: com.google.cloud.tools.jib.plugins.common.BuildStepsExecutionException: Build image failed
        at com.google.cloud.tools.jib.plugins.common.BuildStepsRunner.build(BuildStepsRunner.java:274)
        at com.google.cloud.tools.jib.gradle.BuildImageTask.buildImage(BuildImageTask.java:126)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        ... 144 more
Caused by: java.nio.file.AccessDeniedException: C:\Users\mathijs.degroot\AppData\Local\google-cloud-tools-java\jib\tmp\7311393973508473734 -> C:\Users\mathijs.degroot\AppData\Local\google-cloud-tools-java\jib\layers\12f9f62c1cab33b02b4f332738b03fbcd35fd7906f6fa5ad1acb733056d51cb7
        at com.google.cloud.tools.jib.cache.DefaultCacheStorageWriter.moveIfDoesNotExist(DefaultCacheStorageWriter.java:74)
        at com.google.cloud.tools.jib.cache.DefaultCacheStorageWriter.write(DefaultCacheStorageWriter.java:135)
        at com.google.cloud.tools.jib.cache.DefaultCacheStorage.write(DefaultCacheStorage.java:74)
        at com.google.cloud.tools.jib.cache.Cache.writeCompressedLayer(Cache.java:68)
        at com.google.cloud.tools.jib.builder.steps.PullAndCacheBaseImageLayerStep.call(PullAndCacheBaseImageLayerStep.java:93)
        at com.google.cloud.tools.jib.builder.steps.PullAndCacheBaseImageLayerStep.call(PullAndCacheBaseImageLayerStep.java:37)
        at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:127)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
        at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:80)
        Suppressed: java.nio.file.FileSystemException: C:\Users\mathijs.degroot\AppData\Local\google-cloud-tools-java\jib\tmp\7311393973508473734: failed to delete one or more files; see suppressed exceptions for details
                at com.google.common.io.MoreFiles.throwDeleteFailed(MoreFiles.java:836)
                at com.google.common.io.MoreFiles.deleteRecursively(MoreFiles.java:582)
                at com.google.cloud.tools.jib.filesystem.TemporaryDirectory.close(TemporaryDirectory.java:56)
                at com.google.cloud.tools.jib.cache.DefaultCacheStorageWriter.$closeResource(DefaultCacheStorageWriter.java:101)
                at com.google.cloud.tools.jib.cache.DefaultCacheStorageWriter.write(DefaultCacheStorageWriter.java:148)
                at com.google.cloud.tools.jib.cache.DefaultCacheStorage.write(DefaultCacheStorage.java:74)
                at com.google.cloud.tools.jib.cache.Cache.writeCompressedLayer(Cache.java:68)
                at com.google.cloud.tools.jib.builder.steps.PullAndCacheBaseImageLayerStep.call(PullAndCacheBaseImageLayerStep.java:93)
                at com.google.cloud.tools.jib.builder.steps.PullAndCacheBaseImageLayerStep.call(PullAndCacheBaseImageLayerStep.java:37)
                at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:127)
                at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
                at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:80)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
                at java.lang.Thread.run(Thread.java:748)
                Suppressed: java.nio.file.DirectoryNotEmptyException: C:\Users\mathijs.degroot\AppData\Local\google-cloud-tools-java\jib\tmp\7311393973508473734
                        at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:266)
                        at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
                        at java.nio.file.Files.delete(Files.java:1126)
                        at com.google.common.io.MoreFiles.deleteRecursivelyInsecure(MoreFiles.java:710)
                        at com.google.common.io.MoreFiles.deleteRecursively(MoreFiles.java:571)
                        ... 13 more


* Get more help at https://help.gradle.org

BUILD FAILED in 1m 6s
2 actionable tasks: 1 executed, 1 up-to-date

```

**Environment**: 

**`jib-gradle-plugin` Configuration**: 
```groovy
plugins {
    // Apply the java-library plugin to add support for Java Library
    id 'java-library'
    id 'com.google.cloud.tools.jib' version '1.0.0'
}

jib.to.image = 'madegroot/helloworld'

repositories {
    // Use jcenter for resolving your dependencies.
    // You can declare any Maven/Ivy/file repository here.
    jcenter()
}

dependencies {
    // This dependency is exported to consumers, that is to say found on their compile classpath.
    api 'org.apache.commons:commons-math3:3.6.1'

    // This dependency is used internally, and not exposed to consumers on their own compile classpath.
    implementation 'com.google.guava:guava:26.0-jre'

    // Use JUnit test framework
    testImplementation 'junit:junit:4.12'
}
```
```
D:\repos\personal-repos\gradle-init-project>gradlew --version

------------------------------------------------------------
Gradle 5.1.1
------------------------------------------------------------

Build time:   2019-01-10 23:05:02 UTC
Revision:     3c9abb645fb83932c44e8610642393ad62116807

Kotlin DSL:   1.1.1
Kotlin:       1.3.11
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.13 compiled on July 10 2018
JVM:          1.8.0_162 (Oracle Corporation 25.162-b12)
OS:           Windows 10 10.0 amd64
```
```
D:\repos\personal-repos\gradle-init-project>java -version
java version "1.8.0_162"
Java(TM) SE Runtime Environment (build 1.8.0_162-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.162-b12, mixed mode)
```
```
D:\repos\personal-repos\gradle-init-project>systeminfo

...
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.17763 N/A Build 17763
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Member Workstation
OS Build Type:             Multiprocessor Free
...
System Manufacturer:       HP
System Model:              HP EliteDesk 800 G3 SFF
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 158 Stepping 9 GenuineIntel ~3600 Mhz
BIOS Version:              HP P01 Ver. 02.18, 8-5-2018
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume2
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC+01:00) Amsterdam, Berlin, Bern, Rome, Stockholm, Vienna
Total Physical Memory:     32.647 MB
Available Physical Memory: 7.852 MB
Virtual Memory: Max Size:  38.345 MB
Virtual Memory: Available: 5.409 MB
Virtual Memory: In Use:    32.936 MB
Page File Location(s):     C:\pagefile.sys
...
Hotfix(s):                 4 Hotfix(s) Installed.
                           [01]: KB4469041
                           [02]: KB4470788
                           [03]: KB4471331
                           [04]: KB4469342
Network Card(s):           3 NIC(s) Installed.
                           [01]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (Default Switch)
                                 DHCP Enabled:    No
                                 ...
                           [02]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (DockerNAT)
                                 DHCP Enabled:    No
                                 ...
                           [03]: Intel(R) Ethernet Connection (5) I219-LM
                                 Connection Name: Ethernet
                                 DHCP Enabled:    Yes
                                 ...
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

**Log output**: 
- see output above

**Additional Information**: 
- Tested with multiple projects and on other hardware.
- The jib works fine when executed the same code in Windows Subsystem for Linux (Ubuntu)
```
mathijs@workstation:/$ cd /mnt/d/repos/personal-repos/gradle-init-project/
mathijs@workstation:/mnt/d/repos/personal-repos/gradle-init-project$ ./gradlew jib -Djib.to.auth.username=madegroot -Djib.to.auth.password=QQQ
To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/5.1.1/userguide/gradle_daemon.html.
Daemon will be stopped at the end of the build stopping after processing

Containerizing application to madegroot/helloworld...
Base image 'gcr.io/distroless/java' does not use a specific image digest - build may not be reproducible

Container entrypoint set to [java, -cp, /app/resources:/app/classes:/app/libs/*, jvm.library.Library]

Built and pushed image as madegroot/helloworld
Executing tasks:
[==============================] 100.0% complete


BUILD SUCCESSFUL in 13s
2 actionable tasks: 2 executed
```
```
mathijs@workstation:/mnt/d/repos/personal-repos/gradle-init-project$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.1 LTS
Release:        18.04
Codename:       bionic
```  

This issue possible related to: 
- #1036
- #1037
- #1044
- #1274
Raising a new issue, as there is not much activity on this topic and this is a blocking issue for my project to move forward with this tool.

**Comment by briandealwis** [comment]
`MoreFiles.deleteRecursivelyInsecure` verifies there were no exceptions deleting the directory contents before trying to delete the directory itself.  Seems odd to get a `DirectoryNotEmptyException` on a recursive delete.

@madegroot does this error happen consistently?

**Comment by briandealwis** [comment]
@madegroot Are you running multiple jib builds in parallel?  Could you please check what files remain in that temp directory and their permissions/acls (`C:\Users\mathijs.degroot\AppData\Local\google-cloud-tools-java\jib\tmp\7311393973508473734` in your example above)?

**Comment by madegroot** [comment]
@briandealwis This behavior is constant, all tries on Windows have failed with this same error.
Attempts by colleagues have the same result (Similar hardware. Same OS, Gradle and Java version)

There are 18 sub-directories in `C:\Users\mathijs.degroot\AppData\Local\google-cloud-tools-java\jib\tmp\` left over from previous attempts, all directories are empty, including `7311393973508473734`

**Comment by briandealwis** [comment]
As a work-around, you can run with `-Djib.useOnlyProjectCache=true` to disable that shared cache.  The layers will still be cached within your project.

**Comment by madegroot** [comment]
Unfortunately no succes with the Java upgrade to **Oracle** JDK 8 _202 and renamed jib directory.
The execution of jib fails consistently, with the same exception as reported above.
```
>java -version
java version "1.8.0_202"
Java(TM) SE Runtime Environment (build 1.8.0_202-b08)
Java HotSpot(TM) 64-Bit Server VM (build 25.202-b08, mixed mode)
```

`-Djib.useOnlyProjectCache=true` does work, thank you for the workaround. 

I do hope to assist you in finding the cause of this issue.

**Comment by briandealwis** [comment]
I just can't reproduce this locally using Server JRE, Gradle 5.1.1.  I wonder if it's something caused by antivirus or some other tool you have running in the background?

Does it happen if you set `jib.baseImageCache` to point to somewhere outside of your `AppData`, like `C:\WINDOWS\TEMP\JIB`?

**Comment by madegroot** [comment]
It works fine on my home computer as well. No problems using jib.
Configuration: Win 10 Pro, Gradle 5.1.1, Java 1.8.0_162-b12 and Java 1.8.0_202-b08

I'm going to do some more tests on the corporate workstation, indeed with the corporate virus scanner as prime suspect.

Thank you for raising #1439, this will help with the adoption of Jib in the organisation.

**Comment by briandealwis** [comment]
Let us know here what you're able to figure out!

**Comment by madegroot** [comment]
The problem is caused by the McAfee On-Access Scanner service.
This service is showing abnormally high load, right at the failure point.
Deactivating McAfee On-Access Scanner service in that moment will cause the build is complete successfully. 

Let me know if you want the version information of McAfee in case you want to pursue this problem.

Using the `jib.baseImageCache` environment variable provides an acceptable workaround for now, as not all volumes are actively scanned, but scanned periodically.
Our build server is Linux so that will work fine.

The ` jib.baseImageCache` property is not made available by the jib Gradle Extension as project property.

Implemented workaround for the Windows workstation:
```gradle
if(org.gradle.internal.os.OperatingSystem.current().isWindows()) {
    System.setProperty('jib.baseImageCache', 'D:\\temp\\jib')
}
```

It would be nicer to set the property directly as project property:
```gradle
if(org.gradle.internal.os.OperatingSystem.current().isWindows()) {
    jib.baseImageCache='D:\\temp\\jib'
}
```

This solution will be cleaner when expending the jib configuration by incorporating authentication and/or custom image names and on-premise registries.

**Comment by briandealwis** [comment]
These Jib properties are environmental properties and so aren't suitable for putting into a buildscript — for example, your `jib.baseImageCache` won't work on a Windows box without a D: drive, or for a Linux box.  You could place your workarounds in your developers' `~/.gradle/init.gradle` file.

I'm going to close this issue as it isn't a Jib problem, and we now have a workaround in place with #1439.

**Comment by madegroot** [comment]
@briandealwis Thanks for the quick responses and putting in the effort to support us users!
(Even when they are Windows users ;-)

**Comment by adelinor** [comment]
> Does it happen if you set `jib.baseImageCache` to point to somewhere outside of your `AppData`, like `C:\WINDOWS\TEMP\JIB`?

Using the system property `jib.baseImageCache` to point a different folder resolved this issue for me. Thanks for that.
